rotary/rotary.py

```diff
@@ -0,0 +1,328 @@
+
+from rotary.sample import SequencingFile, Sample
+from rotary.utils import check_for_files, get_cli_arg_path
+
+yaml = YAML()
+rotary_config_name = 'config.yaml'
+
+
+def main():
+    """
+    Collects input arguments and selects a command to perform.
+    """
+
+    """
+    When installing through pip with pyprojects.toml a new python script is generated
+    that calls main() out of rotary.py. Run parser inside main() so it can be called 
+    externally as a function.
+    """
+    parser = parse_cli()
+    args = parser.parse_args()
+
+    output_dir_path = get_cli_arg_path(args, 'output_dir')
+    config_path = get_cli_arg_path(args, 'config')
+    database_dir_path = get_cli_arg_path(args, 'database_dir')
+
+    if hasattr(args, 'jobs'):
+        jobs = args.jobs
+    else:
+        jobs = None
+
+    if hasattr(args, 'snakemake_args'):
+        snakemake_args = args.snakemake_args
+        if snakemake_args:
+            snakemake_args = snakemake_args.split()
+    else:
+        snakemake_args = None
+
+    if hasattr(args, 'force'):
+        override_existing_files = args.force
+    else:
+        override_existing_files = None
+
+    try:
+        os.makedirs(output_dir_path, exist_ok=True)
+    except TypeError:
+        # Catches error that occurs if you run just 'rotary' rather than 'rotary -h' in the CLI.
+        parser.print_help()
+        sys.exit()
+
+    if not config_path: # If no config is specified via CLI.
+        if hasattr(args, 'run'):
+            # If the run sub-command was used, use the config that is in the output dir.
+            config_path = os.path.join(output_dir_path, 'config.yaml')
+        else:
+            # Else get the default config that is saved with this python package.
+            config_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config.yaml')
+
+    with open(config_path) as config_file:
+        config = yaml.load(config_file)
+
+    if database_dir_path:
+        config['db_dir'] = database_dir_path
+    else:
+        config['db_dir'] = os.path.join(output_dir_path, 'databases')
+
+    config['threads'] = os.cpu_count()
+    config['memory'] = round(psutil.virtual_memory().total / (1024 ** 3))
+
+    # Check for the presence of the run configuration files in the output directory.
+    run_files = ['config.yaml', 'samples.tsv']
+    run_file_presences = check_for_files(directory_path=output_dir_path,
+                                         file_names=run_files)
+
+    if run_file_presences == [False, False]:
+        has_run_files = False
+    elif run_file_presences == [True, True]:
+        has_run_files = True
+    else:
+        raise ValueError('The output directory is missing a run configuration file: {}'.format(zip(run_files,
+                                                                                                   run_file_presences)))
+    existing_message = 'Existing run configuration files {} in the output directory.'.format(run_files)
```

Comment:
This isn't necessary to change, but are you aware of f-strings? They were introduced in Python 3.6 and are a more concise way to format text strings. See https://docs.python.org/3.9/tutorial/inputoutput.html#tut-f-strings and https://peps.python.org/pep-0498/
I recently discovered them and really like them for their readability.

Reply:
@jmtsuji Nice, I didn't know about this. I have modified most of them. Seems like a good way to make the code less wordy.